### ISSUE TITLE
[BUG] namespace unavailable for logging secret config

### ISSUE THREAD

**Issue by abonillabeeche** [harvester/harvester#6107]
**Describe the bug**
When creating a secret for Logging, output/flow expects a secret but the secret should exist in cattle-logging-system, but one cannot create secrets from Harvester in the cattle-logging-system namespace. I had to go into the embedded Rancher to see that namespace.

**To Reproduce**
1. Enable Logging AddOn
2. Using the UI, configure output/flows. It expects a secret, but the needed namespace is not available in the UI to create a secret.

**Expected behavior**
The  cattle-logging-system namespace should be available in the UI

**Environment**
 - Harvester ISO version: 1.3.0

**Comment by Yu-Jack** [comment]
Not sure this is a bug or not. But, indeed, harvester filters most namespaces. @bk201 do you know why we filter those namespaces? Is it for security reason?

![image](https://github.com/harvester/harvester/assets/6960289/23386792-4d1e-4d7f-84ab-83f047c9713d)

**Comment by a110605** [comment]
As @Yu-Jack mentioned, harvester dashboard filters out `system ns` from `GET /v1/harvester/namespaces`, only shows non-system ns to user.  See [filterNamespace()](https://github.com/a110605/harvester-dashboard/blob/a0ef90a5a050190e7a9355df4a67a1cdd194120e/pkg/harvester/store/harvester-store/getters.ts#L112)


<img width="1496" alt="Screenshot 2024-07-16 at 11 36 52 AM" src="https://github.com/user-attachments/assets/3af6f37a-2092-4a60-ad34-b21940725a41">

**Comment by a110605** [comment]
Although namespaces selector excluded system ns in options, user can still input `cattle-logging-system` by clicking `Create a new Namespace` in dropdown and choose secrets in `cattle-logging-system` ns on UI.

Of course, user can also edit yaml to input `cattle-logging-system` secrets.


<img width="1496" alt="Screenshot 2024-07-17 at 11 20 34 AM" src="https://github.com/user-attachments/assets/7799c350-16e6-4b4f-9448-30910b084ca9">

**Comment by bk201** [comment]
>  @bk201 do you know why we filter those namespaces? Is it for security reason?

This ensures user won't accidentally manipulate any resource in system namespaces. @abonillabeeche The method provided by Andy in https://github.com/harvester/harvester/issues/6107#issuecomment-2232302609 works for me, can you try it?

**Comment by abonillabeeche** [comment]
I don't think that makes sense at all. The goal is to create a secret in the expected namespace. Can we add visibility to that namespace in that section, or simply enable that namespace only? Or, allow to edit all namespace secrets in the Secrets section.

If we can't, then we should consider the option to document in the [Logging](https://docs.harvesterhci.io/v1.4/logging/harvester-logging/) section, where to place the secret (since that isn't mentioned either) and provide the steps to type the namespace.

**Comment by a110605** [comment]
@Yu-Jack , let's discuss if this UI change (may affect all create pages) or just modify the doc https://docs.harvesterhci.io/v1.4/logging/harvester-logging/

**Comment by Yu-Jack** [comment]
> This ensures user won't accidentally manipulate any resource in system namespaces.
> 
> [@Yu-Jack](https://github.com/Yu-Jack) , let's discuss if this UI change (may affect all create pages) or just modify the doc https://docs.harvesterhci.io/v1.4/logging/harvester-logging/

@a110605  If preventing those system namespace secret from being edited is necessary, we should keep this until we have more feedback. Or we just only make `cattle-logging-system` namespace available in this page.

**Comment by a110605** [comment]
Need a bug fix from rancher/dashboard https://github.com/rancher/dashboard/pull/15544

**Comment by a110605** [comment]
We need to bump rancher/shell to new version to include the fix from rancher/dashboard.
https://github.com/rancher/dashboard/pull/15544

**Comment by noahgildersleeve** [comment]
Validated in v1.7.0-rc3 on single node libvirt/KVM ipxe-examples deploy with elastic logging server. Secret was able to be 

> **Describe the bug** When creating a secret for Logging, output/flow expects a secret but the secret should exist in cattle-logging-system, but one cannot create secrets from Harvester in the cattle-logging-system namespace. I had to go into the embedded Rancher to see that namespace.
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
> 1. Enable Logging AddOn
> 2. Using the UI, configure output/flows. It expects a secret, but the needed namespace is not available in the UI to create a secret.
> 

<img width="2269" height="1081" alt="Image" src="https://github.com/user-attachments/assets/9ba0954b-c50a-4eb6-bef1-eb9e0660a95f" />
